### ISSUE TITLE
Release 0.9.0

### ISSUE THREAD

**Issue by Flamefire** [Return-To-The-Roots/s25client#1088]
It's over time for the next stable. Let's do it!

Proposed plan:

- Check unassigned (to milestone) issues if any are that urgent that they block the release
- Check issues for milestone 0.9.0 if they can be postponed
- Fix open issues
- Create a PR or vote here once everything is ready to double-check that nothing was missed
- Release

**Comment by stefson** [comment]
* the s25edit submodule doesn't have an obvious license, according to the readme it seems to be gplv2 and thus it should have a copy included into it's source tree. 
* please have a tarball with static url available, for distros to fetch. pushing out a 0.9.0_rc1 this way would make testing of packing a lot easier. 
* clang-9.0.0-rc1 has been branched a few days ago, seems like a good oportunity to do a test compile 
* same goes for gcc-9.2.0, it's only a few days or so away from release 
* do a compile and runtime test for the arm port

(assign any of the last three to me if you want to)

**Comment by Flamefire** [comment]
- I created a PR for the editor with a GPL3 license and updated readme: https://github.com/Return-To-The-Roots/s25edit/pull/19 @Flow86 please double check
- There already is: E.g. https://github.com/Return-To-The-Roots/s25client/archive/0.8.2-16.zip
- Postponed till after this release and once they are actually released. Please file an issue after release if they need fixing (I hope they don't)
- I don't think we can test ARM as we don't have that on any CI. It "should" work though as only standard, portable C++ is used. Feel free to test and report bugs.

**Comment by stefson** [comment]
I think the editor needs at least a small readme. It should explain some basics as in which folder the files are saved and loaded from, where to copy them for runtime testing, and maybe instructions on how to send in maps via pullrequests.

**Comment by Flamefire** [comment]
>  It should explain some basics as in which folder the files are saved and loaded from

It doesn't usually matter for the user as it "just works". Nothing to copy, just save and load in RttR, both use the same folders.

I wouldn't want to encourage users to send in maps via PRs. More work for us... We rather need devs.

**Comment by stefson** [comment]
hmm, maybe it's more useable for windows? For me on linux, the editor's file picker dialog looks like this: 

![s25edit](https://user-images.githubusercontent.com/14232936/63273294-02da6800-c29e-11e9-9a3f-bcf573e3962e.png)

and that's all, only relative path to unknown file names in an unknown directory. Path must be typed in, no preview of directory. 

is it /home/user/.s25rttr/WORLDS/ ?

**Comment by Flamefire** [comment]
> is it /home/user/.s25rttr/WORLDS/ ?

Yes. Well ok, so the problem is the missing directory list. Working on that.

**Comment by stefson** [comment]
Uhm, I just noticed that the zip's from the github archive don't include the subdirs, they're all empty folders with nothing in them. Propably true for anything that get's pulled in from an external repo.

**Comment by Flamefire** [comment]
@Flow86 Maybe use travis to tar up a full source tree and upload as an assert via https://docs.travis-ci.com/user/deployment/releases/? How are you currently creating the Gitub releases?

**Comment by Spikeone** [comment]
I will check my map change PR and probably just keep that one map which is used for testing. Also adding 3 new maps for the release :)

**Comment by stefson** [comment]
finally, I managed to get things going on arm: 
```
dtoverlay=vc4-kms-v3d,cma-192
gpu_mem_1024=64
```

![Unbenannt](https://user-images.githubusercontent.com/14232936/67624401-6057db80-f830-11e9-8f08-a463e71e1c0d.png)

**Comment by stefson** [comment]
Some of the new plugin checkboxes and their description's haven't yet been added to the po files.

**Comment by stoecker** [comment]
Any progress? The last release is now more than 2 years old.

**Comment by Flamefire** [comment]
There are a few yet to be fixed things. See the milestone

**Comment by stefson** [comment]
I just emerged clang-9.0.1, there are no regressions. Maybe I'm going to feel silly and try a preview of llvm/clang-10 during the holidays? :-)

**Comment by stefson** [comment]
last updates pulled in new boost-1.72.0, there are no regressions.

**Comment by stefson** [comment]
good news, boost-1.73.0 final has been released and I will emerge it this weekend for extensive testing

**Comment by Flamefire** [comment]
Closing this in favor of the milestone as the definite source of TODOs

**Comment by Mailaender** [comment]
Where can the release be downloaded?

**Comment by stefson** [comment]
I'm using this topic to leave you feedback: there's no compile error with clang-12.0.0-rc2 as of recent git head.
